[dnm] kvserver: more logging for #100096

tbg opened the pull request:
See https://github.com/cockroachdb/cockroach/issues/99464#issuecomment-1482912491 and https://github.com/cockroachdb/cockroach/issues/100096.

Run via gceworker:

```
gh pr checkout -f 99482
ROACHTEST_ALWAYS_COLLECT_ARTIFACTS=true ./pkg/cmd/roachtest/roachstress.sh -c 1 copyfrom/crdb-nonatomic/sf=1/nodes=1
```

Then look at logs from artifacts.

Epic: none
Release note: None

tbg committed:
roachtest: set copy_from_retries_enabled in copyfrom test

See https://github.com/cockroachdb/cockroach/issues/99464#issuecomment-1482912491.

Epic: none
Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/99482)
<!-- Reviewable:end -->

cucaroach commented on pkg/cmd/roachtest/tests/copyfrom.go:
```diff
@@ -119,6 +119,10 @@ func runCopyFromCRDB(ctx context.Context, t test.Test, c cluster.Cluster, sf int
 	require.NoError(t, err)
+	// See https://github.com/cockroachdb/cockroach/issues/99464#issuecomment-1482912491.
+	// This setting only has an effect if !atomic.
+	_, err = db.ExecContext(ctx, `ALTER ROLE ALL SET copy_from_retries_enabled = %t`)
```
Shouldn't there be a bool argument here?

tbg committed:
roachtest: add ROACHTEST_ALWAYS_COLLECT_ARTIFACTS env var

When stressing a roachtest, it can be helpful to have the artifacts collected
even when the test passes, for example to check for markers of an interesting
issue[^1] in the logs.

[^1]: https://github.com/cockroachdb/cockroach/issues/99464#issuecomment-1485715481

Epic: none
Release note: None

tbg committed:
kvserver: log on ErrProposalDropped

Epic: none
Release note: None

tbg committed:
wip

tbg committed:
wip

tbg committed:
kvserver: touch up raft ready handling log

The bytes printed after "wrote" where the append bytes only, this was
confusing. Consolidate. Also, no need to print whether it's sync or
non-blocking-sync again because we already printed that in the timing section.

Found in https://github.com/cockroachdb/cockroach/issues/100096.

Epic: none
Release note: None

tbg committed:
wip

tbg committed:
log more aggressively
